# DAP027

The `Query` APIs allow multiple rows to be returned, which is *fine* - but if you only want a single row, there are
scenario-specific APIs that allow for significant additional optimizations. The
`QueryFirst`, `QueryFirstOrDefault`, `QuerySingle` and `QuerySingleOrDefault` work as you would expect, and compare directly
to the corresponding LINQ [`First`](https://learn.microsoft.com/dotnet/api/system.linq.enumerable.first),
[`FirstOrDefault`](https://learn.microsoft.com/dotnet/api/system.linq.enumerable.firstordefault),
[`Single`](https://learn.microsoft.com/dotnet/api/system.linq.enumerable.single)
and [`SingleOrDefault`](https://learn.microsoft.com/dotnet/api/system.linq.enumerable.singleordefault) methods.

Bad:

``` c#
var order = conn.Query<Order>(sql, args).First();
```

Good:

``` c#
var order = conn.QueryFirst<Order>(sql, args);
```