# DAP042

Usage of [`[Column]`](https://learn.microsoft.com/dotnet/api/system.componentmodel.dataannotations.schema.columnattribute)
attribute and `[DbValue]` attributes are conflicting - Dapper will choose either one or another's name override.

This conflict can lead to unexpected behavior, so we recommend to use only one of them.

Bad:

``` csharp
class MyType
{
    [DbValue(Name = "MyOtherOtherName")]
    [UseColumnAttribute]
    [Column("MyOtherName")]
    public int MyThisName { get; set; }
}
```

Good:

``` csharp
class MyType
{
    [DbValue(Name = "MyOtherOtherName")]
    // without [Column] and [UseColumnAttribute]
    public int MyThisName { get; set; }
}
```

Another good:

``` csharp
class MyType
{
    // without [DbValue]
    [UseColumnAttribute]
    [Column("MyOtherName")]
    public int MyThisName { get; set; }
}
```
