# DAP026

It looks like you're using a `Query` operation (which expects rows) to
invoke something that *definitely does not have a query*. This won't work; did you
mean to use an `Execute` API, or fetch something?

Bad:

``` c#
var id = conn.QuerySingle<int>("""
    insert SomeLookupTable (Name)
    values ('North')
    """);
```

Good:

``` c#
var id = conn.QuerySingle<int>("""
    insert SomeLookupTable (Name)
    output inserted.Id
    values ('North')
    """);
```

Also fine:

``` c#
conn.Execute("""
    insert SomeLookupTable (Name)
    values ('North')
    """);
```