# DAP030

When a constant method-level `[RowCountHint(...)]` is specified, it *must* include a positive integer to use as the estimated row count.

Bad:

``` csharp
[RowCountHint]
public void SomeMethod()
{ ... }
```


Also bad:

``` csharp
[RowCountHint(-15)]
public void SomeMethod()
{ ... }
```

Good:

``` csharp
[RowCountHint(100)]
public void SomeMethod()
{ ... }
```
