# DAP007

The [`CommandType`](https://learn.microsoft.com/dotnet/api/system.data.commandtype) in ADO.NET is quite important. If Dapper
can *see* your command-type at compile-time, but knows that it isn't something it understands, then it isn't sure what you
want to do. If you genuinely have a need to use an unknown command-type, maybe [log an issue](https://github.com/DapperLib/DapperAOT/issues),
explaining the context!

Bad:

``` csharp
conn.Execute("somesql", commandType: (CommandType)42);
```

Good:

``` csharp
conn.Execute("somesql", commandType: CommandType.StoredProcedure);
```
