# DAP046

Dapper uses normalization for the type properties, and this can cause problems when the properties are not unique.
Recomendation is to use a uniquelly normalized name for each property, which would not conflict with other normalized properties names.

Bad:

``` c#
public class MyType
{
    public string First_Name { get; set; }
    public string FirstName { get; set; }
}
```

Good:

``` c#
public class MyType
{
    public string FirstName { get; set; }
    public string FirstNaming { get; set; }
}
```
