# DAP024

A single member cannot be marked with both `[DbValue]` and `[RowCount]`; split out the two purposes.

Bad:

``` csharp
[RowCount, DbValue(...)]
public int X {get;set;}
```

Good:

``` csharp
[RowCount]
public int X {get;set;}

[DbValue(...)]
public int Y {get;set;}
```