# DAP040

_Terminology:_ **factory method** is a static method that returns an instance of a type.

Your type has multiple factory methods. Dapper.AOT expects a type to have a single constructor to use with all data.
You might consider marking your preferred constructor with `[ExplicitConstructor]`.
This should *probably* be the one with the most parameters.

Bad:

``` csharp
class MyType
{
    public int A { get; private set; }

    public static MyType Create1(int a) => new MyType { A = a };

    public static MyType Create2(int a) => new MyType { A = a };
}
```

Good:

``` csharp
class MyType
{
    public int A { get; private set; }

    [ExplicitConstructor]
    public static MyType Create1(int a) => new MyType { A = a };

    public static MyType Create2(int a) => new MyType { A = a };
}
```