# DAP031

When a member-level `[RowCountHint(...)]` is specified (to allow a per-call estimated row count hint), it *must not* include a value.

Bad:

``` csharp
[RowCountHint(43)]
public int RowCount {get;set;}
```

Good:

``` csharp
[RowCountHint]
public int RowCount {get;set;}
```
