# DAP023

Only one member on a type can capture the number of rows affected by a command; remove the duplicate.

Bad:

``` csharp
[RowCount]
public int X {get;set;}

[RowCount]
public int Y {get;set;}
```

Good:

``` csharp
[RowCount]
public int X {get;set;}

public int Y {get;set;}
```