# DAP048

[DbString](https://github.com/DapperLib/Dapper/blob/main/Dapper/DbString.cs) causes heap allocations, but achieves the same as 
[DbValueAttribute](https://github.com/DapperLib/DapperAOT/blob/main/src/Dapper.AOT/DbValueAttribute.cs).

Bad:

``` c#
public void DapperCode(DbConnection conn)
{
    var sql = "SELECT COUNT(*) FROM Foo WHERE Name = @Name;";
    var cars = conn.Query<int>(sql,
        new
        {
            Name = new DbString
            {
                Value = "MyFoo",
                IsFixedLength = false,
                Length = 5,
                IsAnsi = true
            }
        });
}
```

Good:

``` c#
public void DapperCode(DbConnection conn)
{
    var sql = "SELECT COUNT(*) FROM Foo WHERE Name = @Name;";
    var cars = conn.Query<int>(sql,
        new MyPoco
        {
            Name = "MyFoo"
        });
}

class MyPoco
{
	[DbValue(Length = 5, DbType = DbType.AnsiStringFixedLength)] // specify properties here
	public string Name { get; set; }
}
```
