# DAP047

Dapper uses normalization for the type fields, and this can cause problems when the fields are not unique.
Recomendation is to use a uniquelly normalized name for each field, which would not conflict with other normalized fields names.

Bad:

``` c#
public class MyType
{
    public string _firstName;
    public string first_name;
}
```

Good:

``` c#
public class MyType
{
    public string firstName;
    public string firstNaming;
}
```
