# DAP045

Dapper.AOT now allows cancellation to be specified via the parameters; however, only a single `CancellationToken` member is supported.

Bad:

``` c#
conn.ExecuteAsync("some_proc", new { id, x = cancellationToken, y = cancellationToken });
```

Good:

``` c#
conn.ExecuteAsync("some_proc", new { id, x = cancellationToken });
```

Also fine:

``` c#
conn.ExecuteAsync("some_proc", cancellationToken);
```