# DAP048

[DbString](https://github.com/DapperLib/Dapper/blob/main/Dapper/DbString.cs)
achieves the same as [DbValueAttribute](https://github.com/DapperLib/DapperAOT/blob/main/src/Dapper.AOT/DbValueAttribute.cs)
on which a lot of features rely.

// TODO 
Bad:

``` c#
public class MyType
{
    public string _firstName;
    public string first_name;
}
```

Good:

``` c#
public class MyType
{
    public string firstName;
    public string firstNaming;
}
```
