# DAP043

If you are looking at `DAP043`, then most probably you wanted to use [`[Column]`](https://learn.microsoft.com/dotnet/api/system.componentmodel.dataannotations.schema.columnattribute)
attribute to override the name of the type member.

However, due to [historical reasons](https://stackoverflow.com/a/77073456) you need to add a "marker attribute" `[UseColumnAttribute]`
to explicitly let Dapper know you want to override the type member's name.

Bad:

``` csharp
class MyType
{
    [Column("MyOtherName")]
    public int MyThisName { get; set; }
}
```

Good:

``` csharp
class MyType
{
    [UseColumnAttribute]
    [Column("MyOtherName")]
    public int MyThisName { get; set; }
}
```
