# DAP032

Only a single member-level `[RowCountHint(...)]` can be specified (to allow a per-call estimated row count hint); remove the duplicate.

Bad:

``` csharp
[RowCountHint]
public int X {get;set;}

[RowCountHint]
public int Y {get;set;}
```

Good:

``` csharp
[RowCountHint]
public int X {get;set;}

public int Y {get;set;}
```
