# DAP035

It looks like you have multiple constructors on a type marked `[ExplicitConstructor]`; that's just confusing!
Pick one, and remove the attribute from the others. This should *probably* be the one with the most parameters.

Bad:

``` csharp
class MyType
{
    [ExplicitConstructor]
    public MyType(int id, string name) { }

    [ExplicitConstructor]
    public MyType(DateTime when, int id, string name) { }
}
```

Good:

``` csharp
class MyType
{
    public MyType(int id, string name) { }

    [ExplicitConstructor]
    public MyType(DateTime when, int id, string name) { }
}
```