# DAP025

It looks like you're using an `Execute` operation (which doesn't expect rows) to
invoke something that *definitely has a query*. Those results will be
silently ignored, which probably isn't what you intended. Did you mean to use
a `Query` API?

Bad:

``` c#
conn.Execute("""
    select Id, Name
    from SomeLookupTable
    """);
```

Good:

``` c#
var rows = conn.Query<MyLookup>("""
    select Id, Name
    from SomeLookupTable
    """);
```