# DAP223

You very rarely want to `delete` all the rows in a table (and when you do: you might consider `truncate`). Maybe be more specific?

Bad:

``` sql
delete
from Accounts
```

Good:

``` sql
delete
from Accounts
where Id = @id
```