# DAP221

If Dapper is binding columns *by name*, then the columns must all have *unique names* - duplicate columns be bound to the final value, with
any earlier values just wasting bandwidth. This *usually* happens as the result of `join` operations, in which case you should
probably alias one of the columns.

Bad:

``` sql
select a.Id, a.Name, b.Name
from -- not shown
```

Good:

``` sql
select a.Id, a.Name, b.Name as [Category]
from -- not shown
```
