# DAP213

You seem to have assigned a value to a scalar that is never used. This probably isn't intentional.

Bad:

``` sql
declare @i int = 42; -- never used
set @i = 15;
select @i;
```

Good:

``` sql
declare @i int;
set @i = 15;
select @i;
```