# DAP229

When performing a `First`-style query and `top` is specified, you should use `top 1`; additional rows will be silently ignored, so they're
just wasting bandwidth.

Bad:

``` sql
select top 10 Id, Name
from Users
```

Good:

``` sql
select top 1 Id, Name
from Users
```