# DAP232

When using a literal `fetch` value, that value should be positive.

Bad:

``` sql
select Id, Name
from Users
order by Name
offset 0 rows
fetch next 0 row only
```

Good:

``` sql
select Id, Name
from Users
order by Name
offset 0 rows
fetch next 1 row only
```
