# DAP218

When performing a multi-row `insert` using `values`, the number of values in each row must match.

Bad:

``` sql
insert SomeTable (Name, Value)
values ('abc', 42), ('def', 63, 'ghi')
```

Good:

``` sql
insert SomeTable (Name, Value, Description)
values ('abc', 42, ''), ('def', 63, 'ghi')
```
