# DAP234

You seem to have a complex expression that be statically evaluated to a constant; maybe don't do that?

Bad:

``` sql
select Id, Number
from Accounts
where Balance = 23 + 5
```

Good:

``` sql
select Id, Number
from Accounts
where Balance = 28
```

If you're trying to show where the number is coming from, maybe a comment instead?

``` sql
select Id, Number
from Accounts
where Balance = 28 -- 23 + 5
```