# DAP220

If Dapper is binding columns *by name*, then the columns must all *have names* - anonymous columns will never be bound, and are just wasting bandwidth.

Bad:

``` sql
select Id, Credit - Debit
from Accounts
```

Good:

``` sql
select Id, Credit - Debit as [Balance]
from Accounts
```
