# DAP216

It is good practice to explicitly specify columns when performing `insert` operations, as the column order isn't *necessarily* always
the same in all environments (for example, a cleanly created database vs a database that has had cumulative change scripts applied
can end up with slightly different column orders).

Bad:

``` sql
insert SomeTable
values (42)
```

Good:

``` sql
insert SomeTable (SomeColumn)
values (42)
```

It also makes your intent clear to the reader, which is valuable by itself.