# DAP238

It looks like you're doing something the hard way...

Bad:

``` sql
a + 0
b / 1
c * 0
d >> 0
e ^ -1
```

Good:

``` sql
a -- anything plus zero is itself
b -- anything divided by one is itself
0 -- anything multiplied by zero is zero
d -- anything right-shifted by zero is zero
~e -- anything xor'd with a full mask is the bitwise complement
```

