# DAP227

When using a literal `top` value (without the `percent` modifier), that value must be an integer.

Bad:

``` sql
select top 23.5 Id, Name, Address
from Users
```

Good:

``` sql
select u.Id, u.Name, a.Address
from Users u
inner join Addresses a
  on a.UserId = u.Id
```