# DAP244

A `SELECT` expression cannot mix aggregate and non-aggregate results;

Fine:

``` sql
select Name
from SomeTable
```

or

``` sql
select COUNT(1)
from SomeTable
```

Invalid:

``` sql
select Name, COUNT(1)
from SomeTable
```

Aggregate functions are [listed here](https://learn.microsoft.com/sql/t-sql/functions/aggregate-functions-transact-sql).