# DAP224

You very rarely want to `delete` all the rows in a table. Maybe be more specific?

Bad:

``` sql
update Accounts
set Balance = 0
```

Good:

``` sql
update Accounts
set Balance = 0
where Id = @id
```