# DAP235

In TSQL, `TOP` cannot be mixed with `OFFSET`; use a `FETCH` clause instead!

Bad:

``` sql
select top 10 Id, Name
from Users
order by Name
offset 0 rows
```

Good:

``` sql
select Id, Name
from Users
order by Name
offset 0 rows
fetch next 10 rows only
```
