# DAP219

Friends don't let friends `select *`. This is fine for inspecting tables in ad-hoc tools, but is bad practice in application code.
In addition to fetching unnecessary columns (which can impact performance, especially if the columns have large clob/blob data),
the *order* of columns can be unpredictable (see also [DAP216](DAP216)).

Bad:

``` sql
select * from SomeTable
```

Good:

``` sql
select Id, Name, Description from SomeTable
```
