# DAP233

When using a literal `offset` value, that value should be non-negative.

Bad:

``` sql
select Id, Name
from Users
order by Name
offset -1 rows
fetch next 1 row only
```

Good:

``` sql
select Id, Name
from Users
order by Name
offset 0 rows
fetch next 1 row only
```