# DAP217

The number of columns *specified* in an `insert` must match the number of *values* specified.

Bad:

``` sql
insert SomeTable (Name, Value)
values ('abc', 42, 'def')
```

Good:

``` sql
insert SomeTable (Name, Value, Description)
values ('abc', 42, 'def')
```
