# DAP205

Nothing compares to `null`; not even `null`. Nothing is ever *equal* to `null`, but also nothing is ever *not equal* to `null`; **all comparisons return false** (at least, unless you're running with exotic configuration options).

To correctly compare to `null`, you should use [`is [not] null`](https://learn.microsoft.com/sql/t-sql/queries/is-null-transact-sql).

Bad:

``` sql
select A, B
from SomeTable
where X = null
and Y <> null
```

Good:

``` sql
select A, B
from SomeTable
where X is null
and Y is not null
```

